На соревновании по биатлону один из трех спортсменов стреляет и попадает в мишень. 
Вероятность попадания для первого спортсмена равна 0.9, для второго — 0.8, для третьего — 0.6. 
Найти вероятность того, что выстрел произведен:

a) первым спортсменом;

б) вторым спортсменом;

в) третьим спортсменом.


Рассмотрим следующие события:

$A$ - в результате выстрела, цель поражена;

$B_1$ - выстрел произведен первым спортсменом;

$B_2$ - выстрел произведен вторым спортсменом;

$B_3$ - выстрел произведен третьим спортсменом;

Вероятности того, что выстрел произведен определенным спортсменом равны, т.к. выстрел 1, и произвести его могут только 3 этих спортсмена, т.е. события $B_1 , B_2, B_3$ составляют полную группу событий. $\Rightarrow $ $P(B_1)=P(B_2)=P(B_3)=\frac{1}{3}$.

Вероятность того, что произошло событие $A$, при условии наступления события $B$ даны в условии задачи, т.е. $P_{B_1}(A)=0.9, P_{B_2}(A)=0.8, P_{B_3}(A)=0.6$ и и, по формуле полной вероятности, вероятность наступления события $A$ - $P(A)=P(B_1)\cdot P_{B_1}(A)+P(B_2)\cdot P_{B_2}(A)+P(B_3)\cdot P_{B_3}(A)=\frac{1}{3}\cdot 0.9+\frac{1}{3}\cdot 0.8+\frac{1}{3}\cdot 0.6$

In [1]:
PB=1/3
PA=PB*0.9+PB*0.8+PB*0.6
print(f'Полная вероятность наступления события А  = {PA: .4f}')

Полная вероятность наступления события А  =  0.7667


С помощью формулы Байеса оценим вероятность того, что имело место одно из событий $B_n$, т.к. событие $A$ уже произошло

$P_A(B_n)=\frac{P(B_n)\cdot P_{B_n}(A)}{P(A)}\Rightarrow\\P_A(B_1)=\frac{\frac{1}{3}\cdot 0.9}{0.7667}\\P_A(B_2)=\frac{\frac{1}{3}\cdot 0.8}{0.7667}\\P_A(B_1)=\frac{\frac{1}{3}\cdot 0.6}{0.7667}$

In [2]:
PAB1=PB*0.9/PA
PAB2=PB*0.8/PA
PAB3=PB*0.6/PA
print(f'ОТВЕТ:\n'
        f' а) Вероятность того, что выстрел произвёл первый спортсмен: {PAB1: .3f};\n'
        f' б) Вероятность того, что выстрел произвёл второй спортсмен: {PAB2: .3f};\n'
        f' в) Вероятность того, что выстрел произвёл третий спортсмен: {PAB3: .3f}.'
     )

ОТВЕТ:
 а) Вероятность того, что выстрел произвёл первый спортсмен:  0.391;
 б) Вероятность того, что выстрел произвёл второй спортсмен:  0.348;
 в) Вероятность того, что выстрел произвёл третий спортсмен:  0.261.
